middle_layer/main.py

```diff
@@ -55,14 +60,42 @@ def exec(info: ExecInfo):
             with open(f"{tmp_dir_path}/{key}", "w") as data_file:
                 data_file.write(value)
 
+        if len(info.check_code) > 0:
+            # Check code
+            # About --cap-drop: https://www.redhat.com/en/blog/secure-your-containers-one-weird-trick
+            docker_command = f"timeout -s 9 {info.timeout_seconds}s docker run --rm --user $(id -u):$(id -g) --ulimit cpu={info.timeout_seconds} --cpus {cpus} --memory={info.memory_allowed_mb}m --cap-drop=ALL --log-driver=none --workdir /sandbox -v {tmp_dir_path}/:/sandbox/ {info.image_name}:latest /sandbox/code /sandbox/check_code True ''"
```

Comment:
This is closer to what I was wanting. But I would like for us to only use a single `docker run` command, too. The back-end exec scripts would accept a path to the checking code file, test code file, and student's code file. If there is checking code, the exec script would read the checking code file and perform the checks. If the code passes the checks, the exec script would execute the student's code and the test code (if there is test code). Otherwise, it would stop executing at that point.

We will no longer need to pass a boolean argument that indicates whether there is checking code. If there is no checking code, don't create the checking code file (in middle_layer/main.py). Just pass an empty string (in quotes) to the back end. In the exec scripts (back ends), you will need to check whether the path to the checking code is an empty string. If it is, then no checks are necessary. If it is not an empty string, then run the check code.

Please follow the same pattern for the test code. If the user has specified test code, pass write it to the test code file and pass the path to that file to the back end. If not, pass an empty string.